web3.js/test/bpf-loader.test.ts

```diff
@@ -147,6 +147,30 @@ if (process.env.TEST_LIVE) {
         );
       });
 
+      it('simulate transaction with returnData', async () => {
+        const simulatedTransaction = new Transaction().add({
+          keys: [
+            {pubkey: payerAccount.publicKey, isSigner: true, isWritable: true},
+          ],
+          programId: program.publicKey,
+        });
+        const {err, returnData} = (
+          await connection.simulateTransaction(simulatedTransaction, [
+            payerAccount,
+          ])
+        ).value;
+        const expected_return_data = new Uint8Array([1, 2, 3]);
+        var decoded_data = new Uint8Array(
+          atob(returnData.data[0])
+            .split('')
+            .map(function (c) {
+              return c.charCodeAt(0);
+            }),
+        );
```

Comment:
no need to parse for ourselves when our libraries are happy to do it for us :smile: 
```suggestion
        const decodedData = Buffer.from(returnData.data[0], returnData.data[1]);
```